### PR TITLE
Merge list archived workflows into workflow list command

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

- added a flag `--archived` to `tctl worfklow list`
- removed separate command `tctl workflow listarchived`

## Why?

- merging commands after after discussions here https://github.com/temporalio/tctl/pull/162 and on slack

## Checklist

1. Closes 

2. How was this tested:

updated unit tests & executing manually

3. Any docs updates needed?

### CODE REVIEW

File: cli/app_test.go [hunk 1]
```diff
@@ -397,7 +397,7 @@ func (s *cliAppSuite) TestListWorkflow_Open_WithQuery() {
 
 func (s *cliAppSuite) TestListArchivedWorkflow() {
 	s.sdkClient.On("ListArchivedWorkflow", mock.Anything, mock.Anything).Return(&workflowservice.ListArchivedWorkflowExecutionsResponse{}, nil).Once()
-	err := s.app.Run([]string{"", "--namespace", cliTestNamespace, "workflow", "listarchived", "--query", "some query string"})
+	err := s.app.Run([]string{"", "--namespace", cliTestNamespace, "workflow", "list", "--archived", "--query", "some query string"})
```
Comment (bergundy) [inline cli/app_test.go:400]: We could really use some integration tests here but obviously not blocking this PR

File: cli/flags.go [hunk 2]
```diff
@@ -375,6 +377,11 @@ var flagsForWorkflowFiltering = []cli.Flag{
 		Aliases: FlagListQueryAlias,
 		Usage:   FlagListQueryUsage,
 	},
+	&cli.BoolFlag{
+		Name:    FlagArchive,
+		Aliases: FlagArchiveAlias,
+		Usage:   "List archived workflow executions",
```
Comment (bergundy) [inline cli/flags.go:383]:
```suggestion
		Usage:   "List archived workflow executions (EXPERIMENTAL)",
```

Comment (bergundy) [reply cli/flags.go:383]: Why did you resolve this?

Comment (feedmeapples) [reply cli/flags.go:383]: was not intentional. Wanted to commit the suggestion

**Review by bergundy — APPROVED** [review]
Awesome!

**Pull request merged**